tools/ci/src/main.rs

```diff
@@ -1,19 +1,47 @@
-use xshell::cmd;
+use std::io::BufRead;
+
+use duct::cmd;
 
 fn main() {
     // When run locally, results may from actual CI runs triggered by .github/workflows/ci.yml
     // - Official CI runs latest stable
     // - Local runs use whatever the default Rust is locally
 
     // See if any code needs to be formatted
-    cmd!("cargo fmt --all -- --check")
+    println!("$ cargo fmt --all -- --check");
+    let errput = cmd!("cargo", "fmt", "--all", "--", "--check")
+        .stderr_capture()
         .run()
         .expect("Please run 'cargo fmt --all' to format your code.");
+    // Capture stderr, filter out lines starting with "Warning:" that complain about nightly-only
+    // options in rustfmt.toml -- we can remove this (and maybe switch back to `xshell` instead of
+    // `duct`) once either rustfmt.toml doesn't cause warnings in stable or nightly rust, or rustfmt
+    // learns an option to silence warnings provoked by the config file.
+    errput
+        .stderr
+        .lines()
+        .filter_map(|l| l.ok())
+        .filter(|l| !l.starts_with("Warning"))
```

Comment:
Could we make this a bit more specific so that we don't silence all warnings, but just the intended ones due to not running nightly?

Reply:
I'd prefer to stick with filtering by `Warning`.

My concern with more specific filtering is that we may end up with a brittle setup where we have to track the various specific config settings being warned about [as things change](https://github.com/bevyengine/bevy/pull/1479) multiplied across the different Rust versions our users use.

I looked through the `rustfmt` source code and found that `Warning` only occurs on the sorts of config errors we don't care about. I also see that this is the approach they use themselves [in rustfmt's own CI](https://github.com/rust-lang/rustfmt/blob/257311c304fddf578d8d4041ddf129533390b8ef/ci/integration.sh#L54).

A better long-term solution would be to teach `rustfmt` itself a `--no-config-warnings` option.  Or maybe figure out a way to only emit the warning _exactly once_.  Either would be more elegant than dealing with it in the output like we've had to do.

Reply:
I think we should probably just remove the unstable commands from config.toml _and_ the filter here. That way we are maximally safe. We could just comment out the unstable commands for easier access from people (like me) that want the "better" formatting they provide. That feels safer and is makes the implementation easier to grok.

Reply:
✅ Unstable commands removed from `rustfmt.toml`
✅ Filtering commit reverted.